src/backend/distributed/utils/aggregate_utils.c

```diff
@@ -703,20 +791,65 @@ TypecheckWorkerPartialAggArgType(FunctionCallInfo fcinfo, StypeBox *box)
 	Assert(!argtypesNull);
 	ReleaseSysCache(proctuple);
 
-	if (ARR_NDIM(DatumGetArrayTypeP(argtypes)) != 1 ||
-		ARR_DIMS(DatumGetArrayTypeP(argtypes))[0] != 1)
+	if (ARR_NDIM(DatumGetArrayTypeP(argtypes)) != 1)
 	{
 		elog(ERROR, "worker_partial_agg_sfunc cannot type check aggregates "
-					"taking anything other than 1 argument");
+					"taking multi-dimensional arguments");
 	}
 
-	int arrayIndex = 0;
-	Datum argtype = array_get_element(argtypes,
-									  1, &arrayIndex, -1, sizeof(Oid), true, 'i',
-									  &argtypesNull);
-	Assert(!argtypesNull);
+	int naggarg = ARR_DIMS(DatumGetArrayTypeP(argtypes))[0];
```

Comment:
minor: as I noted elsewhere, we mostly prefer more explicit names. So, some suggestions could be `naggarg ` -> `aggregateArgCount`, `idx ` -> `aggregateArgIndex`

Reply:
Makes sense. Will change.